pkg/services/ngalert/state/historian/core.go

```diff
@@ -22,6 +22,12 @@ func shouldRecord(transition state.StateTransition) bool {
 		return false
 	}
 
+	// Do not log transitions when keeping last state
+	toKeepLast := strings.Contains(transition.StateReason, models.StateReasonKeepLast) && !strings.Contains(transition.PreviousStateReason, models.StateReasonKeepLast)
```

Comment:
Shouldn't this be in `ShouldRecordAnnotation` instead? We want to record the transition in state history but not panel annotations, right?